AndroidTestAutomationStarter/app/src/androidTest/java/com/novoda/androidstoreexample/tests/EspressoTestExampleWithMockServer.kt

```diff
@@ -1,60 +1,37 @@
 package com.novoda.androidstoreexample.tests
 
 import android.content.Intent
-import android.support.test.InstrumentationRegistry
 import android.support.test.espresso.Espresso.onView
 import android.support.test.espresso.assertion.ViewAssertions.matches
 import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
 import android.support.test.espresso.matcher.ViewMatchers.withText
-import android.support.test.rule.ActivityTestRule
-import com.novoda.androidstoreexample.EspressoHostModule
 import com.novoda.androidstoreexample.activities.MainActivity
-import com.novoda.androidstoreexample.dagger.App
-import com.novoda.androidstoreexample.dagger.component.DaggerAppComponent
-import com.squareup.okhttp.mockwebserver.MockResponse
-import com.squareup.okhttp.mockwebserver.MockWebServer
+import com.novoda.androidstoreexample.rules.MockServerTestRule
 import org.junit.After
-import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
-import java.io.InputStream
 
 class EspressoTestExampleWithMockServer {
 
-    private val activityTestRule = ActivityTestRule<MainActivity>(MainActivity::class.java, false, false)
-    private val mockWebServer = MockWebServer()
-    private val port = 9091
+    private val activityTestRule = MockServerTestRule(MainActivity::class.java,
+            false,
+            false)
+
 
     @get:Rule
-    var activityRule: ActivityTestRule<MainActivity> = activityTestRule
-
-    @Before
-    @Throws(Exception::class)
-    fun setUp() {
-        val appComponent = DaggerAppComponent.builder().hostModule(EspressoHostModule(port)).build()
-        App.Companion.component = appComponent
-        mockWebServer.play(port)
-    }
+    var activityRule: MockServerTestRule<MainActivity> = activityTestRule
```

Comment:
Should the MockServer and MainActivity know about each other? 
An alternative could be to keep separate rules and chain them as needed, to keep them more flexible too